model/project_parser_db.go

```diff
@@ -54,6 +54,26 @@ func ParserProjectFindOneById(id string) (*ParserProject, error) {
 	return ParserProjectFindOne(ParserProjectById(id))
 }
 
+func ParserProjectFindOneByVersion(id string, version string) (*ParserProject, error) {
+	if version == "" {
+		lastGoodVersion, err := FindVersionByLastKnownGoodConfig(id, -1)
+		if err == nil && lastGoodVersion != nil {
+			version = lastGoodVersion.Id
+		}
+	}
+	if version != "" {
+		parserProject, err := ParserProjectFindOneById(version)
+		if err != nil {
+			return nil, err
+		}
+		if parserProject == nil {
+			return nil, errors.Errorf("Unable to find parser project for project '%s'", id)
+		}
+		return parserProject, nil
+	}
+	return nil, errors.Errorf("Unable to find last good version for project '%s'", id)
```

Comment:
We should error here on 60 instead; it's more clear than waiting to error, and we can wrap the actual problem. Then at 64 we can just assume version != ""